src/inboxsdk-js/loading/platform-implementation-loader.ts

```diff
@@ -1,28 +1,38 @@
-import { PlatformImplementation } from '../../platform-implementation-js/platform-implementation';
+import defer from '../../common/defer';
+import type {
+  PiOpts,
+  PlatformImplementation,
+} from '../../platform-implementation-js/platform-implementation';
 
-const PlatformImplementationLoader = {
-  load(appId: string, opts: any): Promise<PlatformImplementation> {
-    return Promise.resolve()
-      .then(() => {
-        if (!(global as any).__InboxSDKImpLoader) {
-          return PlatformImplementationLoader._loadScript().then(() => {
-            if (!(global as any).__InboxSDKImpLoader) {
-              throw new Error('Implementation file did not load correctly');
-            }
-          });
-        }
-      })
-      .then(() => (global as any).__InboxSDKImpLoader.load('0.1', appId, opts));
-  },
+export class PlatformImplementationLoader {
+  static #loadScriptSet = defer<void>();
```

Comment:
Using the deferred directly as the `loadScript` instead of as a flag would be a little bit more direct.